code/go/0chain.net/blobbercore/challenge/challenge.go

```diff
@@ -70,12 +71,14 @@ func saveNewChallenge(c *ChallengeEntity, ctx context.Context) {
 		}
 	}()
 
-	db := datastore.GetStore().GetDB()
-	if Exists(db, c.ChallengeID) {
+	ctx = datastore.GetStore().CreateTransaction(ctx)
```

Comment:
db.Transaction is enough good to do it. It is unnecessary to box/unbox db transaction in context.Context. It doesn't make memory/db leak.  commit/rollback will be executed automatically.  see https://gorm.io/docs/transactions.html#Transaction

Reply:
Yes they both are the same thing. However, everywhere there is use of context to create and get new transaction.